### PR TITLE
Revamp database appenders

### PR DESCRIPTION
**Pull request by ppkarwasz**
This PR splits the documentation of database appenders into their own files.

As in the previous PRs we:

* check and reformat the configuration options,
* add links to the Plugin Reference,
* improve the readability of the description by shortening it and reformatting it,

Unlike other PRs, since these appenders are rarely used, we don't revamp their configuration examples.

Part of #2528.

### CODE REVIEW

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 1]
```diff
@@ -0,0 +1,1505 @@
...
+
+See <<NoSqlAppender-formatting>> for more details.
+|===
+
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-core_org-apache-logging-log4j-core-appender-nosql-NoSqlAppender[{open-book} Plugin reference for `NoSQL`]
+
+[#NoSqlAppender-formatting]
+=== Formatting
+
+This appender transforms log events into JSON-like structures in two ways:
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc:1065]: I am confused about this `JSON-like structure`. Is it JSON or not? If not, how is it _alike_?

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 2]
```diff
@@ -0,0 +1,1505 @@
...
+| `String`
+| `localhost`
+|
+The host name of the CouchDB server.
+
+| [[CouchDbProvider-attr-port]]port
+| `int`
+a|
+`80`:: for `http`
+`443`:: for `https`
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc:1342]: DL visually looks odd:

```suggestion
`80` (`http`)
`443` (`https`)
```

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 3]
```diff
@@ -0,0 +1,1505 @@
...
+| [[CouchDbProvider-attr-password]]password
+| `String`
+|
+|
+The password for authentication.
+
+4+h| [[CouchDbProvider-attr-factory]]Factory method configuration attributes
+
+| [[CouchDbProvider-attr-factoryClassName]]factoryClassName
+| https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html[`Class<?>`]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc:1367]: Mind removing Javadoc links to basic standard classes like `Class`, please?

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 4]
```diff
@@ -0,0 +1,1505 @@
...
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-core_org-apache-logging-log4j-core-net-SocketAddress[{open-book} Plugin reference for `SocketAddress`]
+
+[#CassandraAppender-examples]
+=== Configuration examples
+
+Here is an example Cassandra Appender configuration:
+
+[source,xml]
+----
+<Configuration name="CassandraAppenderTest">
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc:322]: This needs to be ported to `examples` in all supported configuration formats.

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 5]
```diff
@@ -0,0 +1,1505 @@
...
+      <AppenderRef ref="databaseAppender"/>
+    </Root>
+  </Loggers>
+</Configuration>
+----
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="error">
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc:779]: This needs to be ported to `examples` in all supported configuration formats.

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 6]
```diff
@@ -0,0 +1,1505 @@
...
+</Configuration>
+----
+
+The following configuration uses no layout to indicate that the appender should match the keys of a `MapMessage` to the names of
+``ColumnMapping``s when setting the values of the Appender's SQL INSERT statement.
+This lets you insert rows for custom values in a database table based on a Log4j `MapMessage` instead of values from `LogEvent`.
+
+[source,xml]
+----
+<Configuration status="debug">
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc:805]: This needs to be ported to `examples` in all supported configuration formats.

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 7]
```diff
@@ -0,0 +1,1505 @@
...
+
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-jpa_org-apache-logging-log4j-core-appender-db-jpa-JpaAppender[{open-book} Plugin reference for `JPA`]
+
+[#JpaAppender-examples]
+=== Configuration examples
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="error">
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc:978]: This needs to be ported to `examples` in all supported configuration formats.

**Review by vy — CHANGES_REQUESTED** [review]
Great job @ppkarwasz! :100: I can imagine you put great effort into checking each of these appenders in detail and trying out the shared configurations. :exploding_head:

File: src/site/antora/modules/ROOT/pages/manual/layouts.adoc [hunk 8]
```diff
@@ -16,6 +16,7 @@
 ////
 
 = Layouts
+:open-book: &#x1F4D6;
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/layouts.adoc:19]: I want to use the same character as you did, but I don't want to copy/paste it each time.

[the remark this answers is not in the document]
Comment (vy) [reply src/site/antora/modules/ROOT/pages/manual/layouts.adoc:19]: Adding an extra layer of indirection will simply make things more prone to copy-paste errors. Nobody will remember `open-book`, they will look where it is used anyway. Hence, I prefer the unicode character.

File: src/site/antora/modules/ROOT/examples/manual/appenders/database/nosql-log-event.json
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/examples/manual/appenders/database/nosql-log-event.json:1]: Unfortunately it is not possible to use JTL with the NoSQL appender. It only accepts layouts that return a `MapMessage` as an abstraction of NoSQL document.

The serialization is hardcoded in `NoSqlDatabaseManager` and this is something users complain about (see #1751) for example.

File: src/site/antora/modules/ROOT/pages/manual/layouts.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/layouts.adoc:19]: I'd prefer to keep it the way it is.

File: src/site/antora/modules/ROOT/pages/manual/appenders.adoc [hunk 9]
```diff
@@ -224,6 +224,12 @@ See xref:manual/appenders/file.adoc[] for details.
 Rolling file appenders:::
 See xref:manual/appenders/rolling-file.adoc[] for details.
 
+[#database-appenders]
+=== Database appenders
+
+The appenders write log events directly to a database.
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders.adoc:230]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/11fa9ec4f4d2b98aa6e901f9ba0de0d5ef28deb6

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc (hunk 1 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/11fa9ec4f4d2b98aa6e901f9ba0de0d5ef28deb6

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc (hunk 3 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/79d21ae2a4321c7bc516a9fb99689cf7059c1603

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 10]
```diff
@@ -0,0 +1,1505 @@
...
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-couchdb_org-apache-logging-log4j-couchdb-CouchDbProvider[{open-book} Plugin reference for `CouchDb`]
+
+[#NoSqlAppender-examples]
+=== Configuration examples
+
+Here are a few sample configurations for the NoSQLAppender and MongoDB4 provider:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 11]
```diff
@@ -0,0 +1,1505 @@
...
+    <Root level="ALL">
+      <AppenderRef ref="MongoDbAppender" />
+    </Root>
+  </Loggers>
+</Configuration>
+----
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 12]
```diff
@@ -0,0 +1,1505 @@
...
+    </Root>
+  </Loggers>
+</Configuration>
+----
+
+You can define additional fields to log using KeyValuePair elements, for example:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc [hunk 13]
```diff
@@ -0,0 +1,1505 @@
...
+        </Root>
+    </Loggers>
+</Configuration>
+----
+
+Here are a few sample configurations for the NoSQLAppender and CouchDB provider:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc (hunk 4 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc (hunk 5 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc (hunk 6 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

File: src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc (hunk 7 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc]: Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe

**Comment by ppkarwasz** [comment]
I refactored the example, to:

* be available in 4 formats,
* to only show what is interesting.

**Review by vy — APPROVED** [review]

**Pull request merged**